tests/sentry/integrations/slack/test_unfurl.py

```diff
@@ -940,3 +941,81 @@ def test_bar_chart_display_renders_bar_chart(self, mock_generate_chart, _):
         assert len(mock_generate_chart.mock_calls) == 1
 
         assert mock_generate_chart.call_args[0][0] == ChartType.SLACK_DISCOVER_TOTAL_DAILY
+
+    @patch("sentry.integrations.slack.unfurl.discover.client.get")
+    @patch("sentry.integrations.slack.unfurl.discover.generate_chart", return_value="chart-url")
+    def test_bar_chart_interval_with_absolute_date(self, mock_generate_chart, api_mock):
+        url = f"https://sentry.io/organizations/{self.organization.slug}/discover/results/?display=bar&end=2022-09-16T23%3A59%3A59&field=title&field=event.type&field=project&field=user.display&field=timestamp&name=All+Events&query=&sort=-timestamp&start=2022-09-09T00%3A00%3A00&utc=true&yAxis=count%28%29"
+
+        link_type, args = match_link(url)
+
+        if not args or not link_type:
+            raise Exception("Missing link_type/args")
+
+        links = [
+            UnfurlableUrl(url=url, args=args),
+        ]
+
+        start, end = parse_timestamp(links[0].args.get("query").get("start")), parse_timestamp(
+            links[0].args.get("query").get("end")
+        )
+        expected_interval = get_interval_from_range(end - start, False)
```

Comment:
I think we should hardcode the expected intervals here and the test below just so that if someone introduces a breaking change in that function, this test will scream that something is broken instead of replicating the broken behaviour!

Reply:
Got it thanks!